### ISSUE TITLE
F401 false positive and unsafe fix with subclassing generics

### ISSUE THREAD

**Issue by cibere** [astral-sh/ruff#15560]
I'm encountering a scenario where the F401 rule triggers a false positive when subclassing and providing a generic, with ruff also applying a fix that breaks the typing of the class.

Sample Code: 
```py
from __future__ import annotations
from typing import TYPE_CHECKING, Generic, TypeVar

T = TypeVar("T")

if TYPE_CHECKING:
    from collections.abc import Iterable


class Bar(Generic[T]): ...


class Foo(Bar["Iterable"]): ...
```
and when checking this file with ruff:
```
sample.py:6:33: F401 [*] `collections.abc.Iterable` imported but unused
  |
5 | if TYPE_CHECKING:
6 |     from collections.abc import Iterable
  |                                 ^^^^^^^^ F401
  |
  = help: Remove unused import: `collections.abc.Iterable`

Found 1 error.
[*] 1 fixable with the `--fix` option.
```
Using the `--fix` option changes the code to:
```py
from __future__ import annotations
from typing import TYPE_CHECKING, Generic, TypeVar

T = TypeVar("T")

if TYPE_CHECKING:
    pass


class Bar(Generic[T]): ...


class Foo(Bar["Iterable"]): ...
```
which breaks the typing: `"Iterable" is not defined`

**Comment by cibere** [comment]
just saw #9298
